packages/autocannon-ui-frontend/src/components/ResultSet.jsx

```diff
@@ -46,12 +47,25 @@ const useStyles = makeStyles(theme => ({
     '&.Mui-expanded': {
       borderBottom: `1px solid ${theme.palette.divider}`
     }
+  },
+  accordionHeader: {
+    lineHeight: 2.5
   }
 }))
 
-export default function ResultSet({ data }) {
+export default function ResultSet({ data, onChangeSelection }) {
   const classes = useStyles()
 
+  const [isSelected, setIsSelected] = React.useState(false)
+  const onSelectHandler = React.useCallback(
+    e => {
+      e.stopPropagation()
+      onChangeSelection(data, !isSelected)
```

Comment:
Q: wouldn't this be simpler if we used an uncontrolled component for the checkbox?

Reply:
If we are talking about performance, it is a better approach and it should have been used also for RunOptionsForm fields.

Reply:
not performance, simplicity. just asking, I'm not sure

Reply:
In my opinion, is more readable how I implemented it.